### PR TITLE
FEATURE: Add top-topics + tag-topics

### PR DESCRIPTION
**Pull request by carsick**
`top-topics` shows a count of popular topics configured `all, yearly, quarterly, monthly, weekly, daily`
`tag-topics` shows a count of topics from a tag

Consolidated and matched styling with existing `recent-replies` block

### CODE REVIEW

File: desktop/desktop.scss [hunk 1]
```diff
@@ -51,26 +51,70 @@
...
 }
 
-.recent-replies--date {
-  font-size: var(--font-down-1);
-  color: var(--primary-medium);
+// top topics, tag topics
+
+.tag-topics,
+.top-topics {
+  &--topic {
```
Comment (chapoi) [inline desktop/desktop.scss:86]: Could you use BEM according to our guidelines when writing new CSS?

```suggestion
  &__topic {
```

Comment (carsick) [reply desktop/desktop.scss:86]: I can but I was just matching the existing style in the SCSS, I'll have to update all the existing handlebar files to match

Comment (carsick) [reply desktop/desktop.scss:86]: I fear changing the existing CSS names will undo a lot of customizations communities may already have in place. Like `recent-replies--reply` to `recent-replies__reply`. Should I just have it done for the ones I added for now even if it won't match the existing style?

Comment (pmusaraj) [reply desktop/desktop.scss:86]: Good call, let's keep existing classes unchanged.

File: desktop/desktop.scss [hunk 2]
```diff
@@ -51,26 +51,70 @@
...
 
 // Recent replies
-.recent-replies--reply {
-  border-bottom: 1px solid var(--primary-low);
-  padding-bottom: 0.5em;
-  margin-bottom: 1em;
-  display: flex;
-}
+.recent-replies {
+  &--reply {
```
Comment (pmusaraj) [inline desktop/desktop.scss:55]: Just a code styling not, personally, I am not a fan of splitting an element's class name like this. Makes it harder to find in the future.

File: javascripts/discourse/components/tag-topics.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { ajax } from "discourse/lib/ajax";
+import { tracked } from "@glimmer/tracking";
+
+export default class TagTopics extends Component {
+  @tracked tagtopics = null;
+
+  constructor() {
+    super(...arguments);
+    const count = this.args?.params?.count || 5;
+    const tagtopics = "/tag/" + this.args?.params?.tag + ".json";
```
Comment (pmusaraj) [inline javascripts/discourse/components/tag-topics.js:12]: This constant should maybe be called `tagTopicsUrl`, that way it avoids confusion with `this.tagtopics`.

File: javascripts/discourse/components/top-topics.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      results = data.topic_list.topics;
+      this.toptopics = results.slice(0, count);
+      this.toptopics.forEach((topic) => {
+        topic["category"] = Category.findById(topic.category_id);
+      });
+    });
+  }
+
+  willDestroy() {
+    this.toptopics = null;
```
Comment (pmusaraj) [inline javascripts/discourse/components/top-topics.js:26]: Tiny thing, we tend to use camelCase for JS variables. So `topTopics` vs `toptopics`.

**Review by pmusaraj — APPROVED** [review]
Looks good to me, just a few small code style comments.

File: javascripts/discourse/components/tag-topics.js (hunk 3 above)
Comment (carsick) [reply javascripts/discourse/components/tag-topics.js:12]: Thank you for catching this! I used to call it `featured-tag` but changed it to `tag-topics` to match the syntax of `category-topics` but I totally forgot about addresses this variable conflict

File: desktop/desktop.scss (hunk 1 above)
Comment (chapoi) [reply desktop/desktop.scss]: Sorry but I disagree. I'm not asking to change existing classes, but if we go the route of "keep the same style as the old one even for new stuff" we'll never get BEM implemented. 
So I rather use BEM for any new stuff, even if it is a style change from the rest of the file. If we ever get to refactoring, that's less work left.

Comment (pmusaraj) [reply desktop/desktop.scss]: Looks to me like we're in agreement. Using underscores for the new elements is fine with me, I'd rather not edit classes for the existing components in this PR.

**Pull request merged**